lib/igniter/project/application.ex

```diff
@@ -42,6 +42,57 @@ defmodule Igniter.Project.Application do
     end
   end
 
+  defp expand_app_name(zipper) do
+    with :error <- Common.expand_literal(zipper),
+         :error <- expand_attribute(zipper) do
+      :error
+    end
+  end
+
+  defp expand_attribute(zipper) do
+    with {:@, _, [{attr, _, nil}]} <- zipper.node,
+         {:ok, zipper} <- move_up_to_defmodule_body(zipper) do
+      zipper
+      |> Stream.unfold(fn
```

Comment:
Could this be replaced by a `move_left` function with a predicate?

Reply:
It could, but I don't think `move_left` exists yet, unless I somehow missed it.

We could add `move_left` in the other PR, get it merged in, then update this PR to use it.

Reply:
I'd vote for that I think :) Making more ergonomic (and idiomatic) public APIs to let people build with igniter is high priority at the moment, so I'm always looking for opportunities for that.

Reply:
Definitely a nice improvement using `move_left`! 08db3e5ef1b48096135f88caf1a14ca25d98cb79